### PR TITLE
Add rowspan, colspan and alignment to cells in jats table reader

### PR DESCRIPTION
**Pull request by noahmalmed**
### Background 
See [this issue](https://github.com/jgm/pandoc/issues/8408) for background

### Description 
Parse the colspan, rowspan and alignment attributes of Jats table into the Cell node

### CODE REVIEW

File: src/Text/Pandoc/Readers/JATS.hs [hunk 1]
```diff
@@ -273,26 +273,27 @@ parseBlock (Elem e) =
...
+                      let bodyRowElements = case filterChild (named "tbody") e' of
+                                      Just b -> map parseElement $ filterChildren isRow b
+                                      Nothing -> map parseElement $ filterChildren isRow e'
                       let toAlignment c = case findAttr (unqual "align") c of
                                                 Just "left"   -> AlignLeft
                                                 Just "right"  -> AlignRight
                                                 Just "center" -> AlignCenter
                                                 _             -> AlignDefault
+                      let toColSpan element =  fromMaybe 1 $ safeRead $ fromMaybe "1" $ findAttr (unqual "colspan") element
+                      let toRowSpan element =  fromMaybe 1 $ safeRead $ fromMaybe "1" $ findAttr (unqual "rowspan") element
```
Comment (jgm) [inline src/Text/Pandoc/Readers/JATS.hs:291]: We try to keep lines to 80 columns or less.
In this case, you could probably avoid having two `fromMaybe`s:
```
fromMaybe 1 $ findAttr (unqual "colspan") element >>= safeRead
```
(untested). similarly on next line.

File: src/Text/Pandoc/Readers/JATS.hs [hunk 2]
```diff
@@ -303,15 +304,22 @@ parseBlock (Elem e) =
                                                 Just ws' -> let tot = sum ws'
                                                             in  ColWidth . (/ tot) <$> ws'
                                                 Nothing  -> replicate numrows ColWidthDefault
-                      let toRow = Row nullAttr . map simpleCell
-                          toHeaderRow l = [toRow l | not (null l)]
+
+                      let parseCell = parseMixed plain . elContent
+                      let elementToCell element = cell (toAlignment element) (RowSpan $ toRowSpan element) (ColSpan $ toColSpan element) <$> (parseCell element)
```
Comment (jgm) [inline src/Text/Pandoc/Readers/JATS.hs:309]: line length

File: src/Text/Pandoc/Readers/JATS.hs [hunk 3]
```diff
@@ -303,15 +304,22 @@ parseBlock (Elem e) =
...
+                      let rowElementsToCells elements = mapM elementToCell elements
+                      let toRow = fmap (Row nullAttr) . rowElementsToCells
+                          toHeaderRow element = sequence $ [toRow element | not (null element)]
+
+                      parsedRow <- toHeaderRow headRowElements
+                      parsedRows <- mapM toRow bodyRowElements
```
Comment (jgm) [inline src/Text/Pandoc/Readers/JATS.hs:315]: I think this would be clearer to read if it were `headerRow` and `bodyRows`.

**Review by jgm — COMMENTED** [review]
Looks good, with some minor comments!

**Comment by noahmalmed** [comment]
@jgm Thanks for the review! Made those changes for you 👍

**Comment by jgm** [comment]
Great!  I'm wondering if this completely addresses #8408 or only partially.
(It can be merged now either way, I just want to know whether to indicate that it closes #8408 .)

Does JATS have a concept of a table foot, or multiple table bodies, or row heads?  I don't think these are handled yet.

**Comment by noahmalmed** [comment]
Good point @jgm. 

Taking a look at the [jats docs](https://jats.nlm.nih.gov/archiving/tag-library/1.1d1/n-by90.html), it looks like a table can have more than one body and zero or more footer.
> [<tfoot>](https://jats.nlm.nih.gov/archiving/tag-library/1.1d1/n-8az0.html) [Table Footer (NISO JATS table model)](https://jats.nlm.nih.gov/archiving/tag-library/1.1d1/n-8az0.html), zero or one
[<tbody>](https://jats.nlm.nih.gov/archiving/tag-library/1.1d1/n-g4m0.html) [Table Body (NISO JATS table model)](https://jats.nlm.nih.gov/archiving/tag-library/1.1d1/n-g4m0.html), one or more

Which the pandoc code partially handles. [Here](https://github.com/noahmalmed/pandoc/blob/f19c2054636a949247c4ce4ae634e30cd9bf5ce4/src/Text/Pandoc/Readers/JATS.hs#L325) we can see that the code assumes there is only one body in the table and [here](https://github.com/noahmalmed/pandoc/blob/f19c2054636a949247c4ce4ae634e30cd9bf5ce4/src/Text/Pandoc/Readers/JATS.hs#L326) we can see that the code skips parsing the footer elements. 


It may make sense to just create a new issue to fix the above two issues and include col/row span and alignment parsing in that issue. I'd be happy to write that up if that makes sense to you.

**Comment by jgm** [comment]
Yes, a new issue/PR would be great.
I'll merge this meanwhile!

**Pull request merged**